### PR TITLE
Add Java 21 blog post

### PR DESCRIPTION
**Pull request by mosabua**

### CODE REVIEW

File: _posts/2023-11-03-java-21.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+layout: post
+title: "Help us bringing Java 21 to Trino"
```
Comment (martint) [inline _posts/2023-11-03-java-21.md:3]: "Trino is moving to Java 21"

File: _posts/2023-11-03-java-21.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+author: "Manfred Moser"
+excerpt_separator: <!--more-->
+image: /assets/images/logos/java-duke-21.png
+show_pagenav: false
+---
+
+We have done upgrades to newer Java LTS versions successfully before when we
+upgraded to Java 11 and then [Java 17]({% post_url
+2022-07-14-trino-updates-to-java-17 %}). Each time the improvements on the JVM
+runtime, the garbage collectors, the involved libraries and the dependencies
```
Comment (colebow) [inline _posts/2023-11-03-java-21.md:13]:
```suggestion
runtime, the garbage collectors, the involved libraries, and the dependencies
```

File: _posts/2023-11-03-java-21.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+---
+layout: post
+title: "Help us bringing Java 21 to Trino"
+author: "Manfred Moser"
+excerpt_separator: <!--more-->
+image: /assets/images/logos/java-duke-21.png
+show_pagenav: false
+---
+
+We have done upgrades to newer Java LTS versions successfully before when we
```
[the remark this answers is not in the document]
Comment (colebow) [reply _posts/2023-11-03-java-21.md:10]: Maybe, "We're excited to announce that as of Trino 432, we're now using Java 21."

File: _posts/2023-11-03-java-21.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+In September, [Java 21 was
+released](https://blogs.oracle.com/java/post/the-arrival-of-java-21) as the
+newest long-term support version. The [consolidated release
+notes](https://www.oracle.com/java/technologies/javase/21all-relnotes.html) are
+truly impressive when it comes to breath and depth of improvements throughout
+the runtime, the standard libraries, the included tools, and the overall system.
+
+Java 21 provides numerous great opportunities to improve Trino. Even without
+many code changes the performance benefits can have a significant impact on the
```
Comment (colebow) [inline _posts/2023-11-03-java-21.md:32]:
```suggestion
many code changes, the performance benefits can have a significant impact on the
```

File: _posts/2023-11-03-java-21.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+Taking it one step further, and into the codebase and used libraries, we are
+able to move our performance work to the next level. [Project
+Hummingbird](https://github.com/trinodb/trino/issues/14237), our performance
+fine-tuning initiative, is buzzing already. [Dain Sundstrom](
+https://github.com/dain) shipped some great improvements recently again. Just
+like with our Java 17 upgrade, [Mateusz Gajewski](https://github.com/wendigo)
+has been of critical importance to pull all the necessary changes together.
+
+With the [Trino 432
+release](https://trino.io/docs/current/release/release-432.html) we now made the
```
Comment (colebow) [inline _posts/2023-11-03-java-21.md:44]:
```suggestion
release](https://trino.io/docs/current/release/release-432.html) we have now made the
```

File: _posts/2023-11-03-java-21.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+a next big step. The Trino Docker image was changed to use the Eclipse Temurin
+distribution of Java 21. We have been running our test suites with Java 21 for
+quite some time and all looks good. With this release you are now able to easily
```
Comment (colebow) [inline _posts/2023-11-03-java-21.md:47]:
```suggestion
quite some time and all looks good. With this release, you are now able to easily
```

File: _posts/2023-11-03-java-21.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+test Trino with Java 21. Just use the Docker container in your deployment or
+testing with your own pipeline or with the [Trino Helm
+charts](https://github.com/trinodb/charts).
+
+Our plan is to make Java 21 the required runtime, and move towards adopting the
```
Comment (colebow) [inline _posts/2023-11-03-java-21.md:52]:
```suggestion
Our plan is to make Java 21 the required runtime and move towards adopting the
```

File: _posts/2023-11-03-java-21.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+new language features and libraries. However, before we do that we want your
```
Comment (colebow) [inline _posts/2023-11-03-java-21.md:53]:
```suggestion
new language features and libraries. However, before we do that, we want your
```

File: _posts/2023-11-03-java-21.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dain) [reply _posts/2023-11-03-java-21.md:10]: If we want to be more specific we could say "running with Java 21 by default"

[the remark this answers is not in the document]
Comment (mosabua) [reply _posts/2023-11-03-java-21.md:10]: Updated

**Review by mosabua — COMMENTED** [review]
All updated to feedback.

**Review by nineinchnick — APPROVED** [review]

File: _posts/2023-11-03-java-21.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
+---
+layout: post
+title: "Trino is moving to Java 21"
+author: "Manfred Moser"
+excerpt_separator: <!--more-->
+image: /assets/images/logos/java-duke-21.png
+show_pagenav: false
+---
+
+We're excited to announce that as of version 432, Trino is running with Java 21
```
Comment (martint) [inline _posts/2023-11-03-java-21.md:10]: We may want to clarify that this is for the Trino Docker image

Comment (mosabua) [reply _posts/2023-11-03-java-21.md:10]: How about the updated wording?

**Review by martint — APPROVED** [review]

**Pull request merged**